### PR TITLE
Skip box hunts entirely when at box limit

### PR DESCRIPTION
**Pull request by skyleronken**
also updated common.lic to include a count_boxes helper function.

### CODE REVIEW

**Comment by skyleronken** [comment]
the previous PR would run you to the hunting area even if you were at your box limit. this one optimizes so we dont waste time running. .

**Comment by skyleronken** [comment]
@rpherbig 

Script has been reviewed, tested and confirmed by @izawwlgood 

Ready for merge!

**Comment by izawwlgood** [comment]
I'm Commander Shepard, and I approve of this message.

File: hunting-buddy.lic [hunk 1]
```diff
@@ -40,7 +40,12 @@ class HuntingBuddy
 
     @hunting_info = format_hunting_info(hunting_info)
   end
-
+  
+  def need_boxes?
+    return false unless @settings.box_loot_limit
+    count_boxes(@settings) < @settings.box_loot_limit
```
Comment (rpherbig) [inline hunting-buddy.lic]: Does this rummage once per minute?

File: hunting-buddy.lic [hunk 2]
```diff
@@ -68,6 +73,11 @@ class HuntingBuddy
       # Execute scripts to run before the hunt
       execute_actions(before_actions)
 
+      if stop_on_boxes && !need_boxes?
```
Comment (rpherbig) [inline hunting-buddy.lic]: I'm not a fan of having the logic in two places, but it makes sense as written.

File: common.lic [hunk 3]
```diff
@@ -181,6 +181,14 @@ module DRC
     rummage('B', container)
   end
 
+  def count_boxes(settings)
```
Comment (rpherbig) [inline common.lic]: If you're up for it, this would be great as a separate PR. Otherwise you're going to have people getting errors and having to restart dependency manually.

File: hunting-buddy.lic (hunk 2 above)
Comment (skyleronken) [reply hunting-buddy.lic]: yeah. we toyed with this for a while. One is checking within the context of the current combat-trainer instance. The other is checking globally. Thought about passing as combat-trainer argument, but it didnt really solve anything.

File: common.lic (hunk 3 above)
Comment (skyleronken) [reply common.lic]: sure, let me try that

**Comment by skyleronken** [comment]
@rpherbig should be good. split into two PRs.

**Comment by rpherbig** [comment]
@skyleronken I'll merge this tonight (2 days after the common changes were merged). In the mean time, can you fix the merge conflict in `hunting-buddy`?

**Comment by skyleronken** [comment]
@rpherbig conflict resolved
